pkg/controller/launch/pool_manager.go

```diff
@@ -303,3 +307,58 @@ func (cpm *ConcretePoolManager) healthyAndSchedulable() (healthy int, schedulabl
 	}
 	return
 }
+
+func (cpm *ConcretePoolManager) sortByUnschedulableNodes(nodeIDs []string) []string {
+	nodeLister, err := cpm.nodeObservatory.GetListerForKluster(cpm.Kluster)
+	if err != nil {
+		return nil
+	}
+
+	nodes, err := nodeLister.List(labels.Everything())
+	if err != nil {
+		return nil
+	}
+
+	nodeIDsCopy := make([]string, len(nodeIDs))
+	copy(nodeIDsCopy, nodeIDs)
+
+	openstackIDs := []string{}
+
+	for pos, nodeID := range nodeIDsCopy {
+		found := false
+
+		for _, node := range nodes {
+			id := strings.Replace(node.Spec.ProviderID, "openstack:///", "", 1)
+			if nodeID == id {
+				found = true
+
+				if node.Spec.Unschedulable {
+					nodeIDs = toTop(nodeIDs, pos)
+				}
+
+				break
+			}
+
+		}
+
+		if !found {
+			openstackIDs = append(openstackIDs, nodeID)
+		}
+	}
+
+	// sort abandoned openstack nodes to top
+	for _, oID := range openstackIDs {
+		pos := slices.Index(nodeIDs, oID)
+		if pos > 0 {
+			nodeIDs = toTop(nodeIDs, pos)
+		}
+	}
```

Comment:
Pseudo code for sort function
```suggestion
        kubernetesIDs := map[string]Node
	sort.Slice(nodeIDs, func(i, y int) bool {
		if _, ok := kubernetesIDs[nodeIDs[i]]; !ok {
			return true
		}
		if _, ok := kubernetesIDs[nodeIDs[j]]; !ok {
			return false
		}
		if n, ok := kubernetesIDs[nodeIDs[i]]; ok && n.Spec.Unschedulable {
			return true
		}
		if n, ok := kubernetesIDs[nodeIDs[j]]; ok && n.Spec.Unschedulable {
			return false
		}
		return true
	})
```